contacts/views.py

```diff
@@ -82,3 +85,45 @@ def post(self, request, pk):
         contact = Contact.objects.get(id=pk)
         contact.delete()
         return HttpResponseRedirect(reverse('contacts:contact'))
+
+
+class LoginView(View):
+    form_class = LoginForm
+    template_name = 'contacts/login.html'
+
+    def get(self, request):
+        form = self.form_class()
+        return render(request, self.template_name, {'form': form})
+
+    def post(self, request):
+        form = self.form_class(request.POST)
+        if form.is_valid():
+            un = form.cleaned_data['username']
+            pw = form.cleaned_data['password']
+            user = authenticate(username=un, password=pw)
+            if user:
+                if user.is_active:
+                    login(request, user)
+                    return HttpResponseRedirect(reverse('contacts:contact'))
+                else:
+                    print("Invalid account")
+                    return HttpResponseRedirect(reverse('contacts:user_login'))
+            else:
+                print("The username and password were incorrect")
+                return HttpResponseRedirect(reverse('contacts:user_login'))
+        else:
+            return render(request, self.template_name, {'form': form})
+
+
+class HomePageView(View):
+    template_name = 'contacts/home.html'
+
+    def get(self, request):
+        return render(request, self.template_name)
+
+
+class LogoutView(View):
+    template_name = 'contacts/home.html'
+
+    def get(self, request):
```

Comment:
is it working? we're not logging user out here!

Reply:
Done